api/src/utilities/unit-groups-transformations.ts

```diff
@@ -0,0 +1,277 @@
+      b.unitTypes.sort((e, f) => e.numBedrooms - f.numBedrooms)[0].numBedrooms,
+  );
+
+  sortedUnitGroups?.forEach((group) => {
+    let rentAsPercentIncomeRange: MinMax,
+      rentRange: MinMax,
+      amiPercentageRange: MinMax;
+    group.unitGroupAmiLevels.forEach((level) => {
+      if (
+        level.monthlyRentDeterminationType ===
+        MonthlyRentDeterminationTypeEnum.flatRent
+      ) {
+        rentRange = setMinMax(rentRange, level.flatRentValue);
+      } else {
+        rentAsPercentIncomeRange = setMinMax(
+          rentAsPercentIncomeRange,
+          level.percentageOfIncomeValue,
+        );
+      }
+
+      amiPercentageRange = setMinMax(amiPercentageRange, level.amiPercentage);
+    });
+    const groupSummary: UnitGroupSummary = {
+      unitTypes: group.unitTypes
+        .sort((a, b) => (a.numBedrooms < b.numBedrooms ? -1 : 1))
+        .map((type) => type.name),
+      rentAsPercentIncomeRange,
+      rentRange: rentRange && {
+        min: rentRange.min ? `$${rentRange.min}` : '',
+        max: rentRange.max ? `$${rentRange.max}` : '',
+      },
+      amiPercentageRange,
+      openWaitlist: group.openWaitlist,
+      unitVacancies: group.totalAvailable,
+      bathroomRange: {
+        min: group.bathroomMin,
+        max: group.bathroomMax,
+      },
+      floorRange: {
+        min: group.floorMin,
+        max: group.floorMax,
+      },
+      sqFeetRange: {
+        min: group.sqFeetMin,
+        max: group.sqFeetMax,
+      },
+    };
+    summary.push(groupSummary);
+  });
+
+  return summary;
+};
+
+export const getMaxIncomeAmiChartItems = (
+  amiCharts: AmiChart[] = [],
+): AmiChartItem[] => {
+  if (!amiCharts || amiCharts.length === 0) {
+    return [];
+  }
+
+  if (amiCharts.length === 1) {
+    return amiCharts[0]?.items || [];
+  }
+
+  const allItems = amiCharts.flatMap((chart) => chart?.items || []);
+
+  return allItems.reduce((result: AmiChartItem[], currentItem) => {
+    const existingItemIndex = result.findIndex(
+      (item) =>
+        item.percentOfAmi === currentItem.percentOfAmi &&
+        item.householdSize === currentItem.householdSize,
+    );
+
+    if (existingItemIndex === -1) {
+      result.push(currentItem);
+    } else if (currentItem.income > result[existingItemIndex].income) {
+      result[existingItemIndex] = currentItem;
+    }
+
+    return result;
+  }, []);
+};
+
+// One row for every household size, with max income ranged pulled from all ami charts
+// Used to display the maximum income table
+export const getHouseholdMaxIncomeSummary = (
+  unitGroups: UnitGroup[] = [],
+  amiCharts: AmiChart[],
+): HouseholdMaxIncomeSummary => {
+  const columns = {
+    householdSize: 'householdSize',
+  };
+  const rows = [];
+
+  if (!amiCharts || (amiCharts && amiCharts.length === 0)) {
+    return {
+      columns,
+      rows,
+    };
+  }
+
+  const amiChartItems = getMaxIncomeAmiChartItems(amiCharts);
```

Comment:
For Detroit we have sth like this
```
  // if there are two amiCharts (Detroit only has two), then we can use HUD-MSHDA2021, which is a merge of the two, with the max values of both (HUD had higher values for 30 and 80%)
  const amiChartItems = amiCharts.length === 2 ? HUDMSHDA2021.items : amiCharts[0].items
```
I just created function that will take highest values from each AMI chart.